addons/docs/src/blocks/SourceContainer.tsx

```diff
@@ -1,10 +1,11 @@
 import React, { FC, Context, createContext, useEffect, useState } from 'react';
 import deepEqual from 'fast-deep-equal';
 import { addons } from '@storybook/addons';
-import { StoryId } from '@storybook/api';
+import type { SyntaxHighlighterFormatTypes } from '@storybook/components';
+import type { StoryId } from '@storybook/api';
 import { SNIPPET_RENDERED } from '../shared';
 
-export type SourceItem = string;
+export type SourceItem = [string, SyntaxHighlighterFormatTypes];
```

Comment:
I find array types to be awkward, so unless there's a specific reason we'd prefer that, I'd prefer we introduce a new object. 

```ts
export interface SourceItem {
  code: string,
  format?: string,
}
```

WDYT?